docs/torch/link_pred.rst

```diff
@@ -1,347 +1,402 @@
+	...    edge_index = torch.stack([cols, rows], dim=0)
+	...    h = model(node_features, edge_index)
+	...    return predictor(h[edge_label_index[0]], h[edge_label_index[1]]).squeeze().cpu()
 
-.. code-block:: python
+	>>> def test(model, predictor, split_edge, evaluator, config):
+	...    model.eval()
+	...    predictor.eval()
+	...
+	...    edge_keys = ['train', 'valid', 'valid', 'test', 'test']
+	...    edge_sub_keys = ['edge', 'edge', 'edge_neg', 'edge', 'edge_neg']
+	...
+	...    edge_preds = [[] for _ in range(5)]
+	...
+	...    for i, (key, sub_key) in enumerate(zip(edge_keys, edge_sub_keys)):
+	...        edge_list = split_edge[key][sub_key]
+	...        dataset = create_eval_dataset(edge_list, config)
+	...
+	...        for batch in DataLoader(dataset):
+	...            edge_preds[i].append(evaluate_on_batch(model, predictor, batch))
+	...
+	...        edge_preds[i] = torch.cat(edge_preds[i], dim=0)
+	...
+	...    results = {}
+	...    for K in [10, 50, 100]:
+	...        evaluator.K = K
+	...        results[f'Hits@{K}'] = (
+	...            evaluator.eval({'y_pred_pos': edge_preds[0], 'y_pred_neg': edge_preds[2]})[f'hits@{K}'],
+	...            evaluator.eval({'y_pred_pos': edge_preds[1], 'y_pred_neg': edge_preds[2]})[f'hits@{K}'],
+	...            evaluator.eval({'y_pred_pos': edge_preds[3], 'y_pred_neg': edge_preds[4]})[f'hits@{K}']
+	...        )
+	...
+	...    return results
 
-	>>> MODEL_DIR = f"~/tmp/gat_{np.random.randint(9999999)}"
-	>>> arg_list = [
-	...     "--data_dir", "/tmp/caveman",
-	...     "--mode", "train",
-	...     "--trainer", "base",
-	...     "--backend", "snark",
-	...     "--graph_type", "local",
-	...     "--converter", "skip",
-	...     "--node_type", "0",
-	...     "--feature_idx", "0",
-	...     "--feature_dim", "2",
-	...     "--label_idx", "1",
-	...     "--label_dim", "1",
-	...     "--batch_size", "64",
-	...     "--learning_rate", ".001",
-	...     "--num_epochs", "100",
-	...     "--log_by_steps", "16",
-	...     "--data_parallel_num", "0",
-	...     "--use_per_step_metrics",
-	...     "--model_dir", MODEL_DIR,
-	...     "--metric_dir", MODEL_DIR,
-	...     "--save_path", MODEL_DIR,
-	... ]
-
-
-Train
-=====
-
-Finally we train the model to predict whether an edge exists between any two nodes to via run_dist. We expect the loss to decrease with epochs, the number of epochs and learning rate can be adjusted to better achieve this.
+We can now evaluate our model on the test splits with ogb's evaluator.
 
 .. code-block:: python
 
-	>>> run_dist(
-	...     init_model_fn=create_model,
-	...     init_dataset_fn=create_dataset,
-	...     init_optimizer_fn=create_optimizer,
-	...     init_args_fn=init_args,
-	...     run_args=arg_list,
-	... )
+	>>> from ogb.linkproppred import Evaluator
+	>>> evaluator = Evaluator(name='ogbl-collab',)
+	>>> test(model, predictor, dataset.get_edge_split(), evaluator, config)
```

Comment:
We should add some accuracy validation to ensure model keeps working as expected.